packages/adelus/src/defines.h

```diff
@@ -0,0 +1,200 @@
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef __DEFINESH__
+#define __DEFINESH__
+
+
+#undef CBLAS
+#undef DEBUG
+#define  OVERLAP
+#undef  PRINT_STATUS
+#undef   TIMING0
+
+
+#ifdef DREAL
+#define MPI_DATA_TYPE MPI_DOUBLE
+#define MPI_DATA_TYPE2 MPI_DOUBLE_INT
+#endif
+#ifdef SREAL
+#define MPI_DATA_TYPE MPI_FLOAT
+#define MPI_DATA_TYPE2 MPI_FLOAT_INT
+#endif
+
+/*  ------------------ Define Data Types -------------------------- */
+#ifdef ZCPLX
+    typedef struct {
+      double r;
+      double i;
+    } dcomplex ;
+#define DATA_TYPE dcomplex
+#define COMPLEX
+#define MPI_DATA_TYPE MPI_DOUBLE
+#define MPI_DATA_TYPE2 MPI_DOUBLE_INT
+#endif
+
+#ifdef SCPLX
+    typedef struct {
+      float r;
+      float i;
+    } scomplex ;
+#define DATA_TYPE scomplex
+#define COMPLEX
+#define MPI_DATA_TYPE  MPI_FLOAT
+#define MPI_DATA_TYPE2 MPI_FLOAT_INT
+#endif
+
+#ifdef DREAL
+
+#define DATA_TYPE double
+#undef COMPLEX
+#endif
+
+#ifdef SREAL
+
+#define DATA_TYPE float
+#undef COMPLEX
+#endif
+
+
+/*  ------------------ Define Constants and Operations ------------ */
+
+#ifdef ZCPLX
+
+#define CONST_ONE {1.0, 0.0}
+#define CONST_MINUS_ONE {-1.0, 0.0}
+#define CONST_ZERO {0.0, 0.0}
+#define NEGATIVE(X,Y) (Y).r=-(X).r;(Y).i=-(X).i
+#define ABS_VAL(X) ((X).r * (X).r + (X).i * (X).i)
+#define INVERSE(X,W,Z) (Z).r=(X).r/(W);(Z).i=-(X).i/(W)
+#define MULTIPLY(X,Y,Z) (Z).r=(X).r*(Y).r-(X).i*(Y).i;(Z).i=(X).r*(Y).i+(X).i*(Y).r
+#define DIVIDE(X,Y,W,Z) (Z).r=((X).r*(Y).r+(X).i*(Y).i)/(W);(Z).i=((X).i*(Y).r-(X).r*(Y).i)/(W)
+
+#endif
+
+#ifdef  SCPLX
+
+#define CONST_ONE {1.0, 0.0}
+#define CONST_MINUS_ONE {-1.0, 0.0}
+#define CONST_ZERO {0.0, 0.0}
+#define NEGATIVE(X,Y) (Y).r=-(X).r;(Y).i=-(X).i
```

Comment:
This may be too much for this change, but since these are defines remember that they are defined globally for all projects using your library.  These particular names are very common.  It is common to wrap these in library specific names to prevent collision.  For example, using ADELUS_NEGATIVE instead of NEGATIVE ensures that no-one with their own NEGATIVE is likely to run into this.  For most defines, it is more nice than necessary - but especially those with common names you may consider adding a library specific component to prevent collision.

Reply:
Thanks @SAdamBlake. I added ADELUS_ in front of all of these.